Content-Encoding header is null but not expected

`DropwizardApacheConnector` has the private class BufferedJerseyRequestHttpEntity that expects a `Content-Encoding` header that is not suppose to be mandatory.

https://github.com/dropwizard/dropwizard/blob/release/4.0.x/dropwizard-client/src/main/java/io/dropwizard/client/DropwizardApacheConnector.java#L222 is throwing a `NullPointerException` for me because `clientRequest.getRequestHeader(HttpHeaders.CONTENT_ENCODING)` is return null.

Causing code:

```java
package foo;

import io.dropwizard.client.JerseyClientBuilder;
import io.dropwizard.client.JerseyClientConfiguration;
import io.dropwizard.core.setup.Environment;
import jakarta.ws.rs.client.Client;
import jakarta.ws.rs.client.Entity;

public class Main {

    public static void main(String[] args) {
    
        JerseyClientConfiguration clientConfiguration = new JerseyClientConfiguration();
        clientConfiguration.setGzipEnabledForRequests(false);
    
        Client client = (new JerseyClientBuilder(new Environment("env")))
                .using(clientConfiguration)
                .build("whatever");
    
        client
                .target("https://www.example.com")
                .request()
                .post(Entity.text(""));
    }
}

```

Full stack:
```
Exception in thread "main" jakarta.ws.rs.ProcessingException: java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because the return value of "org.glassfish.jersey.client.ClientRequest.getRequestHeader(String)" is null
	at io.dropwizard.client.DropwizardApacheConnector.apply(DropwizardApacheConnector.java:105)
	at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:300)
	at org.glassfish.jersey.client.JerseyInvocation.lambda$invoke$0(JerseyInvocation.java:662)
	at org.glassfish.jersey.client.JerseyInvocation.call(JerseyInvocation.java:697)
	at org.glassfish.jersey.client.JerseyInvocation.lambda$runInScope$3(JerseyInvocation.java:691)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:205)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:390)
	at org.glassfish.jersey.client.JerseyInvocation.runInScope(JerseyInvocation.java:691)
	at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:661)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:439)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.post(JerseyInvocation.java:345)
	at foo.Main.main(Main.java:23)
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because the return value of "org.glassfish.jersey.client.ClientRequest.getRequestHeader(String)" is null
	at io.dropwizard.client.DropwizardApacheConnector$JerseyRequestHttpEntity.<init>(DropwizardApacheConnector.java:222)
	at io.dropwizard.client.DropwizardApacheConnector.getHttpEntity(DropwizardApacheConnector.java:175)
	at io.dropwizard.client.DropwizardApacheConnector.buildApacheRequest(DropwizardApacheConnector.java:121)
	at io.dropwizard.client.DropwizardApacheConnector.apply(DropwizardApacheConnector.java:87)
	... 13 more
```

--- a/dropwizard-client/src/main/java/io/dropwizard/client/DropwizardApacheConnector.java
+++ b/dropwizard-client/src/main/java/io/dropwizard/client/DropwizardApacheConnector.java
@@ -30,6 +30,7 @@ import java.util.ArrayList;
 import java.util.Optional;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
+import java.util.List;
 
 /**
  * Dropwizard Apache Connector.
@@ -219,10 +220,19 @@ public class DropwizardApacheConnector implements Connector {
         private JerseyRequestHttpEntity(ClientRequest clientRequest) {
             super(
                 clientRequest.getMediaType().toString(),
-                clientRequest.getRequestHeader(HttpHeaders.CONTENT_ENCODING).stream().findFirst().orElse(null),
+                getEncoding(clientRequest),
                 true
             );
             this.clientRequest = clientRequest;
+        }
+
+        @Nullable
+        private static String getEncoding(ClientRequest clientRequest) {
+            List<String> contentEncoding = clientRequest.getRequestHeader(HttpHeaders.CONTENT_ENCODING);
+            if (contentEncoding == null) {
+                return null;
+            }
+            return contentEncoding.stream().findFirst().orElse(null);
         }
 
         /**

--- a/dropwizard-client/src/main/java/io/dropwizard/client/DropwizardApacheConnector.java
+++ b/dropwizard-client/src/main/java/io/dropwizard/client/DropwizardApacheConnector.java
@@ -226,15 +226,6 @@ public class DropwizardApacheConnector implements Connector {
             this.clientRequest = clientRequest;
         }
 
-        @Nullable
-        private static String getEncoding(ClientRequest clientRequest) {
-            List<String> contentEncoding = clientRequest.getRequestHeader(HttpHeaders.CONTENT_ENCODING);
-            if (contentEncoding == null) {
-                return null;
-            }
-            return contentEncoding.stream().findFirst().orElse(null);
-        }
-
         /**
          * {@inheritDoc}
          */
@@ -305,7 +296,7 @@ public class DropwizardApacheConnector implements Connector {
         private BufferedJerseyRequestHttpEntity(ClientRequest clientRequest) {
             super(
                 clientRequest.getMediaType().toString(),
-                clientRequest.getRequestHeader(HttpHeaders.CONTENT_ENCODING).stream().findFirst().orElse(null),
+                getEncoding(clientRequest),
                 false
             );
             final ByteArrayOutputStream stream = new ByteArrayOutputStream(BUFFER_INITIAL_SIZE);
@@ -370,6 +361,15 @@ public class DropwizardApacheConnector implements Connector {
         @Override
         public void close() throws IOException {
         }
+    }
+
+    @Nullable
+    private static String getEncoding(ClientRequest clientRequest) {
+        List<String> contentEncoding = clientRequest.getRequestHeader(HttpHeaders.CONTENT_ENCODING);
+        if (contentEncoding == null) {
+            return null;
+        }
+        return contentEncoding.stream().findFirst().orElse(null);
     }
 }
 

